OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -77,38 +79,32 @@ public override bool Tick(Actor self)
 			if (carryall.State != Carryall.CarryallState.Reserved)
 				return true;
 
-			switch (state)
+			var distance = (cargo.Location - self.Location).Length;
+			if (distance <= 4 && !locked)
```

Comment:
Can we please use the `CenterPos` distances instead?

Reply:
Done.